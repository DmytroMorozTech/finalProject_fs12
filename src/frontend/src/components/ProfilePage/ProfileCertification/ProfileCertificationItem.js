import React from 'react'
import style from './styles'
import CreateIcon from '@material-ui/icons/Create'
import Typography from '@material-ui/core/Typography'
import toggleModalAction from '../../../redux/Modal/modalActions'
import {useDispatch} from 'react-redux'
import convertLocalDateToString from '../../../utils/convertLocalDateToString'
import { EDIT_CERTIFICATION } from '../../../redux/Modal/modalTypes'

function ProfileCertificationItem (props) {
  const {name, expirationDate, issueDate, hasExpiryDate, issuingOrganization, credentialId, credentialUrl} = props.certification

  const classes = style()
  const dispatch = useDispatch()
  const issuedMonthAndYear = convertLocalDateToString(issueDate)
  const expirationMonthAndYear = hasExpiryDate ? convertLocalDateToString(expirationDate) : ''

  return (
    <div className={classes.content}>
      <div className={classes.school}>
        <Typography variant="body1" className={classes.schoolName}>{name}</Typography>
        <Typography>{issuingOrganization}</Typography>
        <Typography>Issued: {issuedMonthAndYear}</Typography>
        <Typography> {hasExpiryDate ? `Expires: ${expirationMonthAndYear}` : 'No Expiration Date'} </Typography>
        <Typography>Credential ID: {credentialId}</Typography>
        <Typography>Credential URL: {credentialUrl}</Typography>
      </div>
      <div className={classes.btnCreate} onClick={() =>
        dispatch(toggleModalAction({modalType: EDIT_CERTIFICATION, certification: props.certification}))
      }>
        <CreateIcon className={classes.createIcon}/>
      </div>
    </div>
  )
}
export default ProfileCertificationItem