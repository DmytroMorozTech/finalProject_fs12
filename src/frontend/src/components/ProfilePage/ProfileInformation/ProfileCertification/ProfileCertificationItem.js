import React from 'react'
import style from '../styles'
import CreateIcon from '@material-ui/icons/Create'
import Typography from '@material-ui/core/Typography'
import toggleModalAction from '../../../../redux/Modal/modalActions'
import {useDispatch} from 'react-redux'
import convertLocalDateToString from '../../../../utils/convertLocalDateToString'
import { EDIT_CERTIFICATION } from '../../../../redux/Modal/modalTypes'
import clsx from 'clsx'

function ProfileCertificationItem (props) {
  const {name, expirationDate, issueDate, hasExpiryDate,
    issuingOrganization, credentialId, credentialUrl} = props.certification
  const {isEditable} = props

  const classes = style()
  const dispatch = useDispatch()
  const issuedMonthAndYear = convertLocalDateToString(issueDate)
  const expirationMonthAndYear = hasExpiryDate ? convertLocalDateToString(expirationDate) : ''

  return (
    <div className={classes.content}>
      <div>
        <Typography variant="body1" className={classes.title}>{name}</Typography>
        <Typography>{issuingOrganization}</Typography>
        <Typography>Issued: {issuedMonthAndYear}</Typography>
        <Typography> {hasExpiryDate ? `Expires: ${expirationMonthAndYear}` : 'No Expiration Date'} </Typography>
        {credentialId && <Typography>Credential ID: {credentialId}</Typography>}
        {credentialUrl && <Typography>Credential URL: {credentialUrl}</Typography>}
      </div>
      <div
        id='toggleModal'
        className={clsx(classes.btnCreate, !isEditable && classes.hidden)}
        onClick={() => dispatch(toggleModalAction(
          {modalType: EDIT_CERTIFICATION, certification: props.certification}
        ))}
      >
        <CreateIcon className={classes.createIcon}/>
      </div>
    </div>
  )
}

export default ProfileCertificationItem
