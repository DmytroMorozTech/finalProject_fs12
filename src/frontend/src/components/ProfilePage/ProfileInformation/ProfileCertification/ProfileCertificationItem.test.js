import { shallow } from 'enzyme'
import * as redux from 'react-redux'
import React from 'react'
import ProfileCertificationItem from './ProfileCertificationItem'

jest.mock('../styles', () => () => ({}))

let isEditable

let component

const setUp = (props) => shallow(<ProfileCertificationItem {...props}/>)

describe('should render ProfileCertificationItem component', () => {
  let spyOnUseDispatch
  let mockDispatch

  let certification = {
    'id': 1,
    'name': 'Spring data jpa',
    'issuingOrganization': 'DAN-IT',
    'hasExpiryDate': true,
    'issueDate': '2016-01-01',
    'expirationDate': '2026-01-01',
    'credentialId': 'PS35092',
    'credentialUrl': 'https://dan-it.com.ua'
  }

  beforeEach(() => {
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    isEditable = false

    component = setUp({
      certification,
      isEditable
    })
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
  it('should work dispatch', () => {
    expect(mockDispatch).toHaveBeenCalledTimes(0)
    component.find('div[id=\'toggleModal\']').simulate('click')
    expect(mockDispatch).toHaveBeenCalledTimes(1)
  })
  it('has expiry date', () => {
    expect.stringContaining('Jan 2026')
  })
})

describe('should render ProfileCertificationItem component', () => {
  let spyOnUseDispatch
  let mockDispatch

  let certification = {
    'id': 2,
    'name': 'On-site practice',
    'issuingOrganization': 'Unilever',
    'hasExpiryDate': false,
    'issueDate': '2017-07-08',
    'expirationDate': null,
    'credentialId': 'RQ3627AS',
    'credentialUrl': 'https://unilever.ua'
  }

  beforeEach(() => {
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    isEditable = true

    component = setUp({
      certification,
      isEditable
    })
  })
  it('has not expiry date', () => {
    expect.stringContaining('No Expiration Date')
  })
})
