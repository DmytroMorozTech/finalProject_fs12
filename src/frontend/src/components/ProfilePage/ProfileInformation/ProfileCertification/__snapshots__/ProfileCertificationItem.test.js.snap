// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ProfileCertificationItem component to match snapshot 1`] = `
<div>
  <div>
    <WithStyles(ForwardRef(Typography))
      variant="body1"
    >
      On-site practice
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))>
      Unilever
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))>
      Issued: 
      Jul 2017
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))>
       
      No Expiration Date
       
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))>
      Credential ID: 
      RQ3627AS
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))>
      Credential URL: 
      https://unilever.ua
    </WithStyles(ForwardRef(Typography))>
  </div>
  <div
    className=""
    id="toggleModal"
    onClick={[Function]}
  >
    <Memo(ForwardRef(CreateIcon)) />
  </div>
</div>
`;
